src/smolagents/monitoring.py

```diff
@@ -29,23 +30,76 @@
 from smolagents.utils import escape_code_brackets
 
 
-__all__ = ["AgentLogger", "LogLevel", "Monitor"]
+__all__ = ["AgentLogger", "LogLevel", "Monitor", "TokenUsage", "Timing"]
+
+
+@dataclass
+class TokenUsage:
+    """
+    Contains the token usage information for a given step or run.
+    """
+
+    input_tokens: int
+    output_tokens: int
+
+    def dict(self):
+        return {
+            "input_tokens": self.input_tokens,
+            "output_tokens": self.output_tokens,
+            "total_tokens": self.total_tokens,
+        }
+
+    @property
+    def total_tokens(self):
+        return self.input_tokens + self.output_tokens
+
+    def __str__(self):
+        attributes = vars(self).copy()
+        attributes["total_tokens"] = self.total_tokens  # This makes sure the total tokens are also printed
+        return f"TokenUsage({', '.join(f'{key}={value}' for key, value in attributes.items())})"
+
+
+@dataclass
+class Timing:
+    """
+    Contains the timing information for a given step or run.
+    """
+
+    start_time: float
+    end_time: float | None = None
+
+    def dict(self):
+        return {
+            "start_time": self.start_time,
+            "end_time": self.end_time,
+            "duration": self.duration,
+        }
+
+    @property
+    def duration(self):
+        if self.end_time is None:
+            return None
+        return self.end_time - self.start_time
+
+    def __str__(self):
+        attributes = vars(self).copy()
+        attributes["duration"] = self.duration  # This makes sure the duration is also printed
+        return f"Timing({', '.join(f'{key}={value}' for key, value in attributes.items())})"
 
 
 class Monitor:
     def __init__(self, tracked_model, logger):
         self.step_durations = []
         self.tracked_model = tracked_model
         self.logger = logger
-        if getattr(self.tracked_model, "last_input_token_count", "Not found") != "Not found":
-            self.total_input_token_count = 0
-            self.total_output_token_count = 0
+        self.total_input_token_count = 0
+        self.total_output_token_count = 0
 
-    def get_total_token_counts(self):
-        return {
-            "input": self.total_input_token_count,
-            "output": self.total_output_token_count,
-        }
+    def get_total_token_counts(self) -> TokenUsage:
+        return TokenUsage(
+            input_tokens=self.total_input_token_count,
+            output_tokens=self.total_output_token_count,
+        )
```

Comment:
For backward compatibility, maybe better:
- keep `get_total_token_counts` as it was implemented before, but raising a deprecation warning
- implement a new `get_token_usage` method?